packages/nuxt/src/core/schema.ts

```diff
@@ -57,17 +59,33 @@ export default defineNuxtModule({
 
     // Watch for schema changes in development mode
     if (nuxt.options.dev) {
+      const onChange = debounce(async () => {
+        schema = await resolveSchema()
+        await writeSchema(schema)
+      })
+
+      if (nuxt.options.experimental.parcelWatcher) {
+        const watcherPath = await tryResolveModule('@parcel/watcher', [nuxt.options.rootDir, ...nuxt.options.modulesDir])
+        if (watcherPath) {
+          const { subscribe } = await import(pathToFileURL(watcherPath).href).then(interopDefault) as typeof import('@parcel/watcher')
+          for (const layer of nuxt.options._layers) {
+            const subscription = await subscribe(layer.config.rootDir, onChange, {
+              ignore: ['!nuxt.schema.*']
+            })
+            nuxt.hook('close', () => subscription.unsubscribe())
+          }
+          return
+        }
+        console.warn('[nuxt] falling back to `chokidar` as `@parcel/watcher` cannot be resolved in your project.')
+      }
+
       const filesToWatch = await Promise.all(nuxt.options._layers.map(layer =>
         resolver.resolve(layer.config.rootDir, 'nuxt.schema.*')
       ))
       const watcher = chokidar.watch(filesToWatch, {
```

Comment:
Shouldn't we disable chokidar when the experiment is enabled?

Reply:
we early return and never hit this statement when parcel watcher can be resolved/imported

Reply:
Code style suggestion: Early return by negative check (warn + return) in first if statement could make it more readable...

Reply:
I don't want to return if parcel watcher can't be found - I just want to warn and then proceed to initialise chokidar